models/user.go

```diff
@@ -629,8 +629,18 @@ func (u *User) GetOrganizations(all bool) error {
 // DisplayName returns full name if it's not empty,
 // returns username otherwise.
 func (u *User) DisplayName() string {
-	if len(u.FullName) > 0 {
-		return u.FullName
+	trimmed := strings.TrimSpace(u.FullName)
+	if len(trimmed) > 0 {
+		return trimmed
+	}
+	return u.Name
+}
+
+// GitName returns a git safe name
+func (u *User) GitName() string {
+	gitName := strings.TrimSpace(strings.NewReplacer("\n", "", "<", "", ">", "").Replace(u.FullName))
```

Comment:
Maybe 
```
.Replace(u.DisplayName())
```
is better.

Reply:
Hmm, well the only benefit of passing in `u.DisplayName()` (at this point) is that, it's guaranteed to have been whitespace trimmed or be equal to `u.Name`.

You have to `TrimSpace` after removing the bad characters as you could have an initial `<` followed by spaces, or a terminal `<` preceded by spaces. So the `TrimSpace` in `u.DisplayName()` isn't actually useful and AFAICS `u.Name` is guaranteed to be of an appropriate form for git's author name as those characters are not allowed in u.Name and is pretrimmed.

So in the end all display name does is create a string that doesn't get used.

Reply:
I have however discovered that `u.Name` isn't guaranteed to not have these characters therefore I've placed a second check on the `u.Name` before declaring the names totally pathological and rendering their name as `"user-$UID"`.